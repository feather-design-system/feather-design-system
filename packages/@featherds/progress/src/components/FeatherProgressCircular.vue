<template>
  <div
    class="feather-progress-container"
    :class="{ ...containerClasses }"
    :aria-busy="!completed"
  >
    <span class="reduced-motion-caption">
      {{ reducedMotionText }}
    </span>
    <template v-if="isIndeterminate">
      <CircularIndeterminate :size="size" />
    </template>
    <template v-if="isDeterminate">
      <CircularDeterminate :size="size" :value="value" />
    </template>
    <progress :max="max" :value="value" />
    <span ref="announce" class="announce" aria-live="assertive"></span>
    <div v-if="hasError" class="error-message">
      <slot name="error">{{ error }}</slot>
    </div>
    <template v-else>
      <div
        v-if="showPercentage && isDeterminate"
        class="feather-progress-report"
        aria-live="off"
      >
        {{ reportText }}
      </div>
    </template>
  </div>
</template>

<script lang="ts">
import CircularDeterminate from "./CircularDeterminate.vue";
import CircularIndeterminate from "./CircularIndeterminate.vue";
import { _setTimeout } from "@featherds/utils/setTimeout";

import {
  computed,
  defineComponent,
  onMounted,
  PropType,
  toRef,
  toRefs,
  watch,
} from "vue";
import { useLabelProperty } from "@featherds/composables/LabelProperty";
import { useProgress } from "@featherds/composables/progress/Progress";
const LABELS = {
  reducedMotion: "Loading...",
  indeterminate: "In progress, please wait",
  units: "${value} percent",
  completed: ", progress complete",
  error: "Progress interrupted, ",
  report: "${value}${symbol}",
  symbol: "%",
};

const props = {
  error: { type: String, required: false },
  showPercentage: { type: Boolean, required: false, default: false },
  size: { type: String, required: false, default: "lg" },
  value: { type: Number, required: false },
  labels: {
    type: Object as PropType<Partial<typeof LABELS>>,
    default: () => {
      return LABELS;
    },
  },
};

export default defineComponent({
  props,
  components: {
    CircularDeterminate,
    CircularIndeterminate,
  },
  setup(props) {
    const max = 100;
    const labels = useLabelProperty<typeof LABELS>(
      toRef(props, "labels"),
      LABELS
    );
    const { error, value } = toRefs(props);
    const {
      announce,
      completed,
      containerClasses,
      hasError,
      isDeterminate,
      isIndeterminate,
      percentage,
    } = useProgress(value, error);

    const reducedMotionText = computed(() => {
      return labels.reducedMotionLabel.value;
    });

    const reportText = computed(() => {
      let reportLabel = labels.reportLabel.value || "";
      let symbolLabel = labels.symbolLabel.value || "";
      const status = reportLabel
        .replace("${symbol}", symbolLabel)
        .replace(
          "${value}",
          percentage.value ? percentage.value.toString() : ""
        );
      return status;
    });

    const statusAnnouncement = computed(() => {
      let unitsLabel = labels.unitsLabel.value || "";
      const valueString = percentage.value ? percentage.value.toString() : "";
      return unitsLabel.replace("${value}", valueString);
    });

    const completedAnnouncement = computed(() => {
      return statusAnnouncement.value + labels.completedLabel.value;
    });

    watch(error, (newValue) => {
      announce.value.innerHTML = newValue
        ? labels.errorLabel.value + newValue
        : "";
    });

    watch(value, (newValue, oldValue) => {
      if (newValue === oldValue) return;
      newValue = newValue ? newValue : 0;

      if (isDeterminate.value) {
        if (newValue >= max) {
          announce.value.innerHTML = completedAnnouncement.value;
        } else {
          announce.value.innerHTML = statusAnnouncement.value;
        }
      }
    });

    onMounted(() => {
      if (isIndeterminate.value) {
        _setTimeout(() => {
          // NOTE: chrome does not announce without the delay; nextTick has no effect.
          announce.value.innerHTML = labels.indeterminateLabel.value;
        }, 250);
      }
    });

    return {
      announce,
      completed,
      containerClasses,
      error,
      hasError,
      isDeterminate,
      isIndeterminate,
      max,
      reducedMotionText,
      reportText,
      value,
    };
  },
});
</script>

<style lang="scss" scoped>
// @use "sass:math";
@import "@featherds/styles/themes/variables";
@import "@featherds/styles/mixins/typography";
@import "@featherds/styles/mixins/elevation";

progress {
  display: none;
}
span.announce {
  display: inline-block;
  overflow: hidden;
  height: 0;
  width: 0;
}

circle {
  stroke-linecap: butt;
  transition: stroke-dashoffset 0.5s ease-in-out;
  stroke: var($background);
  opacity: 0.85;
}
.feather-progress-container {
  position: relative;
  display: grid;
  place-items: center;
  width: 160px;
  .feather-progress-circular {
    justify-items: center;
    margin-bottom: 8px;
  }
  .feather-progress-report {
    @include caption;
    text-align: center;
  }
  &.error .error-message {
    position: relative;
    @include caption;
    text-align: center;
    right: 0;
    color: var($error);
  }

  &.determinate {
    .feather-progress-circular {
      .feather-progress-track {
        &.completed {
          animation: done 1s ease-in-out;
        }
      }
      .feather-progress-indicator {
        stroke: var($secondary);
        &.completed {
          stroke: var($secondary);
          animation: done 1s ease-in-out;
        }
      }
    }
  }
}
@keyframes done {
  0% {
    stroke: var($background);
  }
  100% {
    stroke: var($secondary);
  }
}
@keyframes indeterminate-progress {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
@media screen {
  .feather-progress-container {
    .reduced-motion-caption {
      display: none;
    }
  }
}
@media (prefers-reduced-motion) {
  .feather-progress-container {
    .reduced-motion-caption {
      @include caption;
      position: relative;
      display: block;
    }
    .determinate-container,
    .indeterminate-container {
      display: none;
      height: 0;
      opacity: 0;
    }
  }
}
</style>
