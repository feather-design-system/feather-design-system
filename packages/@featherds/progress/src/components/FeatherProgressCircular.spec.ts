import FeatherProgressCircular from "./FeatherProgressCircular.vue";
import { mount } from "@vue/test-utils";
import axe from "@featherds/utils/test/axe";
// import { _setTimeout } from "@featherds/utils/setTimeout";

import { expect, describe, it, vi } from "vitest";

const getWrapper = (opts: Record<string, unknown>) =>
  mount(FeatherProgressCircular, opts);

describe("FeatherProgressCircular.vue", () => {
  describe("Determinate", () => {
    it("should render as html progress tag", () => {
      const props = { value: 0 };
      const wrapper = getWrapper({ props });
      expect(wrapper.find("progress").exists()).toBe(true);
    });
    it("should have container classified as determinate", () => {
      const props = { value: 0 };
      const wrapper = getWrapper({ props });
      expect(wrapper.element).toMatchSnapshot();
    });
    it("should support large size by default", () => {
      const props = { value: 0 };
      const wrapper = getWrapper({ props });
      expect(wrapper.element).toMatchSnapshot();
    });
    it("should render various sizes", async () => {
      const props = { value: 0 };
      const wrapper = getWrapper({ props });
      // const circular = wrapper.find(".feather-progress-circular");

      await wrapper.setProps({ size: "sm" });
      expect(wrapper.element).toMatchSnapshot();

      await wrapper.setProps({ size: "md" });
      expect(wrapper.element).toMatchSnapshot();

      await wrapper.setProps({ size: "lg" });
      expect(wrapper.element).toMatchSnapshot();
    });
    it("should handle errors", async () => {
      const props = {
        error: "",
        showPercentage: false,
        value: 0,
      };
      const slots = {
        error: "",
      };

      const wrapper = getWrapper({ props, slots });

      await wrapper.setProps({ error: "" });
      expect(wrapper.element).toMatchSnapshot();

      await wrapper.setProps({ error: "Error Message." });
      expect(wrapper.element).toMatchSnapshot();
    });
    it("should calculate strokeDashoffset", async () => {
      const props = {
        value: 0,
      };
      const wrapper = getWrapper(props);
      await wrapper.setProps({ value: 0 });
      expect(wrapper.element).toMatchSnapshot();

      await wrapper.setProps({ value: 100 });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
  describe("Indeterminate", () => {
    it("should render as html progress tag", () => {
      const wrapper = getWrapper({});
      expect(wrapper.find("progress").exists()).toBe(true);
    });
    it("should be classified as indeterminate", () => {
      const wrapper = getWrapper({});
      expect(wrapper.element).toMatchSnapshot();
    });
    it("should support large size by default", () => {
      const wrapper = getWrapper({});
      expect(wrapper.element).toMatchSnapshot();
    });
    it("should render various sizes", async () => {
      const wrapper = getWrapper({});

      await wrapper.setProps({ size: "sm" });
      expect(wrapper.element).toMatchSnapshot();

      await wrapper.setProps({ size: "md" });
      expect(wrapper.element).toMatchSnapshot();

      await wrapper.setProps({ size: "lg" });
      expect(wrapper.element).toMatchSnapshot();
    });
    it("should handle errors", async () => {
      const props = {
        error: "",
      };
      const slots = {
        error: "Error Message.",
      };

      const wrapper = getWrapper({ props, slots });

      await wrapper.setProps({ error: "" });
      expect(wrapper.element).toMatchSnapshot();

      await wrapper.setProps({ error: "Error Message." });
      expect(wrapper.element).toMatchSnapshot();
    });
    it("should announce indeterminate message on mount", async () => {
      vi.useFakeTimers();
      const wrapper = getWrapper({});
      setTimeout(() => {}, 500);
      vi.runAllTimers();
      vi.useRealTimers();
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
describe("a11y", () => {
  it("should be accessible", async () => {
    const props = {
      title: "a11y",
    };
    const wrapper = getWrapper({ props });
    expect(await axe(wrapper.element)).toHaveNoViolations();
  });
});
