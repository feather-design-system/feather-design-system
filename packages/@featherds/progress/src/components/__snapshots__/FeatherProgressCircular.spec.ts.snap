// Vitest Snapshot v1

exports[`FeatherProgressCircular.vue > Determinate > should calculate strokeDashoffset 1`] = `
<div
  aria-busy="true"
  class="feather-progress-container determinate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <!--v-if-->
  <div
    class="determinate-container lg"
  >
    <svg
      class="feather-progress-circular"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="feather-progress-track"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-width="12"
      />
      <circle
        class="feather-progress-indicator"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-dashoffset="552.9203070318035"
        stroke-width="12"
        transform="rotate(-90 50 50)"
      />
    </svg>
  </div>
  <progress
    max="100"
    value="0"
  />
  <span
    aria-live="assertive"
    class="announce"
  >
    0 percent
  </span>
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Determinate > should calculate strokeDashoffset 2`] = `
<div
  aria-busy="false"
  class="feather-progress-container completed determinate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <!--v-if-->
  <div
    class="determinate-container lg"
  >
    <svg
      class="feather-progress-circular"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="feather-progress-track"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-width="12"
      />
      <circle
        class="feather-progress-indicator"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-dashoffset="276.46015351590177"
        stroke-width="12"
        transform="rotate(-90 50 50)"
      />
    </svg>
  </div>
  <progress
    max="100"
    value="100"
  />
  <span
    aria-live="assertive"
    class="announce"
  >
    100 percent, progress complete
  </span>
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Determinate > should handle errors 1`] = `
<div
  aria-busy="true"
  class="feather-progress-container determinate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <!--v-if-->
  <div
    class="determinate-container lg"
  >
    <svg
      class="feather-progress-circular"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="feather-progress-track"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-width="12"
      />
      <circle
        class="feather-progress-indicator"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-dashoffset="552.9203070318035"
        stroke-width="12"
        transform="rotate(-90 50 50)"
      />
    </svg>
  </div>
  <progress
    max="100"
    value="0"
  />
  <span
    aria-live="assertive"
    class="announce"
  />
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Determinate > should handle errors 2`] = `
<div
  aria-busy="true"
  class="feather-progress-container completed error"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <!--v-if-->
  <div
    class="determinate-container lg"
  >
    <svg
      class="feather-progress-circular"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="feather-progress-track"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-width="12"
      />
      <circle
        class="feather-progress-indicator"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-dashoffset="552.9203070318035"
        stroke-width="12"
        transform="rotate(-90 50 50)"
      />
    </svg>
  </div>
  <progress
    max="100"
    value="0"
  />
  <span
    aria-live="assertive"
    class="announce"
  >
    Progress interrupted, Error Message.
  </span>
  <div
    class="error-message"
  >
    
    Error Message.
    
  </div>
</div>
`;

exports[`FeatherProgressCircular.vue > Determinate > should have container classified as determinate 1`] = `
<div
  aria-busy="true"
  class="feather-progress-container determinate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <!--v-if-->
  <div
    class="determinate-container lg"
  >
    <svg
      class="feather-progress-circular"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="feather-progress-track"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-width="12"
      />
      <circle
        class="feather-progress-indicator"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-dashoffset="552.9203070318035"
        stroke-width="12"
        transform="rotate(-90 50 50)"
      />
    </svg>
  </div>
  <progress
    max="100"
    value="0"
  />
  <span
    aria-live="assertive"
    class="announce"
  />
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Determinate > should render various sizes 1`] = `
<div
  aria-busy="true"
  class="feather-progress-container determinate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <!--v-if-->
  <div
    class="determinate-container sm"
  >
    <svg
      class="feather-progress-circular"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="feather-progress-track"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-width="12"
      />
      <circle
        class="feather-progress-indicator"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-dashoffset="552.9203070318035"
        stroke-width="12"
        transform="rotate(-90 50 50)"
      />
    </svg>
  </div>
  <progress
    max="100"
    value="0"
  />
  <span
    aria-live="assertive"
    class="announce"
  />
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Determinate > should render various sizes 2`] = `
<div
  aria-busy="true"
  class="feather-progress-container determinate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <!--v-if-->
  <div
    class="determinate-container md"
  >
    <svg
      class="feather-progress-circular"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="feather-progress-track"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-width="12"
      />
      <circle
        class="feather-progress-indicator"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-dashoffset="552.9203070318035"
        stroke-width="12"
        transform="rotate(-90 50 50)"
      />
    </svg>
  </div>
  <progress
    max="100"
    value="0"
  />
  <span
    aria-live="assertive"
    class="announce"
  />
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Determinate > should render various sizes 3`] = `
<div
  aria-busy="true"
  class="feather-progress-container determinate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <!--v-if-->
  <div
    class="determinate-container lg"
  >
    <svg
      class="feather-progress-circular"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="feather-progress-track"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-width="12"
      />
      <circle
        class="feather-progress-indicator"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-dashoffset="552.9203070318035"
        stroke-width="12"
        transform="rotate(-90 50 50)"
      />
    </svg>
  </div>
  <progress
    max="100"
    value="0"
  />
  <span
    aria-live="assertive"
    class="announce"
  />
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Determinate > should support large size by default 1`] = `
<div
  aria-busy="true"
  class="feather-progress-container determinate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <!--v-if-->
  <div
    class="determinate-container lg"
  >
    <svg
      class="feather-progress-circular"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="feather-progress-track"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-width="12"
      />
      <circle
        class="feather-progress-indicator"
        cx="50"
        cy="50"
        fill="transparent"
        r="44"
        stroke-dasharray="552.9203070318035"
        stroke-dashoffset="552.9203070318035"
        stroke-width="12"
        transform="rotate(-90 50 50)"
      />
    </svg>
  </div>
  <progress
    max="100"
    value="0"
  />
  <span
    aria-live="assertive"
    class="announce"
  />
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Indeterminate > should announce indeterminate message on mount 1`] = `
<div
  aria-busy="true"
  class="feather-progress-container indeterminate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <div
    class="indeterminate-container lg"
  >
    <svg
      viewBox="0 0 100 100"
    >
      <clippath
        id="clip"
      >
        <path
          d="M 50 0 a 50 50 0 0 1 0 100 50 50 0 0 1 1 -100 v 10 a 22 22 0 0 0 0 80 22 22 0 0 0 1 -80"
        />
      </clippath>
      <foreignobject
        clip-path="url(#clip)"
        height="100"
        width="100"
        x="0"
        y="0"
      >
        <div
          class="track"
        />
      </foreignobject>
      <g
        class="indicator"
      >
        <foreignobject
          clip-path="url(#clip)"
          height="100"
          width="100"
          x="0"
          y="0"
        >
          <div
            class="gradient"
          />
        </foreignobject>
        <!-- &lt;circle class=&quot;tip&quot; r=&quot;5.15&quot; cx=&quot;51&quot; cy=&quot;5&quot;&gt;&lt;/circle&gt; -->
        <rect
          class="tip"
          height="10"
          width="3"
          x="50"
        />
      </g>
    </svg>
  </div>
  <!--v-if-->
  <progress
    max="100"
  />
  <span
    aria-live="assertive"
    class="announce"
  >
    In progress, please wait
  </span>
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Indeterminate > should be classified as indeterminate 1`] = `
<div
  aria-busy="true"
  class="feather-progress-container indeterminate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <div
    class="indeterminate-container lg"
  >
    <svg
      viewBox="0 0 100 100"
    >
      <clippath
        id="clip"
      >
        <path
          d="M 50 0 a 50 50 0 0 1 0 100 50 50 0 0 1 1 -100 v 10 a 22 22 0 0 0 0 80 22 22 0 0 0 1 -80"
        />
      </clippath>
      <foreignobject
        clip-path="url(#clip)"
        height="100"
        width="100"
        x="0"
        y="0"
      >
        <div
          class="track"
        />
      </foreignobject>
      <g
        class="indicator"
      >
        <foreignobject
          clip-path="url(#clip)"
          height="100"
          width="100"
          x="0"
          y="0"
        >
          <div
            class="gradient"
          />
        </foreignobject>
        <!-- &lt;circle class=&quot;tip&quot; r=&quot;5.15&quot; cx=&quot;51&quot; cy=&quot;5&quot;&gt;&lt;/circle&gt; -->
        <rect
          class="tip"
          height="10"
          width="3"
          x="50"
        />
      </g>
    </svg>
  </div>
  <!--v-if-->
  <progress
    max="100"
  />
  <span
    aria-live="assertive"
    class="announce"
  />
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Indeterminate > should handle errors 1`] = `
<div
  aria-busy="true"
  class="feather-progress-container indeterminate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <div
    class="indeterminate-container lg"
  >
    <svg
      viewBox="0 0 100 100"
    >
      <clippath
        id="clip"
      >
        <path
          d="M 50 0 a 50 50 0 0 1 0 100 50 50 0 0 1 1 -100 v 10 a 22 22 0 0 0 0 80 22 22 0 0 0 1 -80"
        />
      </clippath>
      <foreignobject
        clip-path="url(#clip)"
        height="100"
        width="100"
        x="0"
        y="0"
      >
        <div
          class="track"
        />
      </foreignobject>
      <g
        class="indicator"
      >
        <foreignobject
          clip-path="url(#clip)"
          height="100"
          width="100"
          x="0"
          y="0"
        >
          <div
            class="gradient"
          />
        </foreignobject>
        <!-- &lt;circle class=&quot;tip&quot; r=&quot;5.15&quot; cx=&quot;51&quot; cy=&quot;5&quot;&gt;&lt;/circle&gt; -->
        <rect
          class="tip"
          height="10"
          width="3"
          x="50"
        />
      </g>
    </svg>
  </div>
  <!--v-if-->
  <progress
    max="100"
  />
  <span
    aria-live="assertive"
    class="announce"
  />
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Indeterminate > should handle errors 2`] = `
<div
  aria-busy="true"
  class="feather-progress-container completed error"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <div
    class="indeterminate-container lg"
  >
    <svg
      viewBox="0 0 100 100"
    >
      <clippath
        id="clip"
      >
        <path
          d="M 50 0 a 50 50 0 0 1 0 100 50 50 0 0 1 1 -100 v 10 a 22 22 0 0 0 0 80 22 22 0 0 0 1 -80"
        />
      </clippath>
      <foreignobject
        clip-path="url(#clip)"
        height="100"
        width="100"
        x="0"
        y="0"
      >
        <div
          class="track"
        />
      </foreignobject>
      <g
        class="indicator"
      >
        <foreignobject
          clip-path="url(#clip)"
          height="100"
          width="100"
          x="0"
          y="0"
        >
          <div
            class="gradient"
          />
        </foreignobject>
        <!-- &lt;circle class=&quot;tip&quot; r=&quot;5.15&quot; cx=&quot;51&quot; cy=&quot;5&quot;&gt;&lt;/circle&gt; -->
        <rect
          class="tip"
          height="10"
          width="3"
          x="50"
        />
      </g>
    </svg>
  </div>
  <!--v-if-->
  <progress
    max="100"
  />
  <span
    aria-live="assertive"
    class="announce"
  >
    Progress interrupted, Error Message.
  </span>
  <div
    class="error-message"
  >
    
    Error Message.
    
  </div>
</div>
`;

exports[`FeatherProgressCircular.vue > Indeterminate > should render various sizes 1`] = `
<div
  aria-busy="true"
  class="feather-progress-container indeterminate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <div
    class="indeterminate-container sm"
  >
    <svg
      viewBox="0 0 100 100"
    >
      <clippath
        id="clip"
      >
        <path
          d="M 50 0 a 50 50 0 0 1 0 100 50 50 0 0 1 1 -100 v 10 a 22 22 0 0 0 0 80 22 22 0 0 0 1 -80"
        />
      </clippath>
      <foreignobject
        clip-path="url(#clip)"
        height="100"
        width="100"
        x="0"
        y="0"
      >
        <div
          class="track"
        />
      </foreignobject>
      <g
        class="indicator"
      >
        <foreignobject
          clip-path="url(#clip)"
          height="100"
          width="100"
          x="0"
          y="0"
        >
          <div
            class="gradient"
          />
        </foreignobject>
        <!-- &lt;circle class=&quot;tip&quot; r=&quot;5.15&quot; cx=&quot;51&quot; cy=&quot;5&quot;&gt;&lt;/circle&gt; -->
        <rect
          class="tip"
          height="10"
          width="3"
          x="50"
        />
      </g>
    </svg>
  </div>
  <!--v-if-->
  <progress
    max="100"
  />
  <span
    aria-live="assertive"
    class="announce"
  />
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Indeterminate > should render various sizes 2`] = `
<div
  aria-busy="true"
  class="feather-progress-container indeterminate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <div
    class="indeterminate-container md"
  >
    <svg
      viewBox="0 0 100 100"
    >
      <clippath
        id="clip"
      >
        <path
          d="M 50 0 a 50 50 0 0 1 0 100 50 50 0 0 1 1 -100 v 10 a 22 22 0 0 0 0 80 22 22 0 0 0 1 -80"
        />
      </clippath>
      <foreignobject
        clip-path="url(#clip)"
        height="100"
        width="100"
        x="0"
        y="0"
      >
        <div
          class="track"
        />
      </foreignobject>
      <g
        class="indicator"
      >
        <foreignobject
          clip-path="url(#clip)"
          height="100"
          width="100"
          x="0"
          y="0"
        >
          <div
            class="gradient"
          />
        </foreignobject>
        <!-- &lt;circle class=&quot;tip&quot; r=&quot;5.15&quot; cx=&quot;51&quot; cy=&quot;5&quot;&gt;&lt;/circle&gt; -->
        <rect
          class="tip"
          height="10"
          width="3"
          x="50"
        />
      </g>
    </svg>
  </div>
  <!--v-if-->
  <progress
    max="100"
  />
  <span
    aria-live="assertive"
    class="announce"
  />
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Indeterminate > should render various sizes 3`] = `
<div
  aria-busy="true"
  class="feather-progress-container indeterminate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <div
    class="indeterminate-container lg"
  >
    <svg
      viewBox="0 0 100 100"
    >
      <clippath
        id="clip"
      >
        <path
          d="M 50 0 a 50 50 0 0 1 0 100 50 50 0 0 1 1 -100 v 10 a 22 22 0 0 0 0 80 22 22 0 0 0 1 -80"
        />
      </clippath>
      <foreignobject
        clip-path="url(#clip)"
        height="100"
        width="100"
        x="0"
        y="0"
      >
        <div
          class="track"
        />
      </foreignobject>
      <g
        class="indicator"
      >
        <foreignobject
          clip-path="url(#clip)"
          height="100"
          width="100"
          x="0"
          y="0"
        >
          <div
            class="gradient"
          />
        </foreignobject>
        <!-- &lt;circle class=&quot;tip&quot; r=&quot;5.15&quot; cx=&quot;51&quot; cy=&quot;5&quot;&gt;&lt;/circle&gt; -->
        <rect
          class="tip"
          height="10"
          width="3"
          x="50"
        />
      </g>
    </svg>
  </div>
  <!--v-if-->
  <progress
    max="100"
  />
  <span
    aria-live="assertive"
    class="announce"
  />
  
  <!--v-if-->
  
</div>
`;

exports[`FeatherProgressCircular.vue > Indeterminate > should support large size by default 1`] = `
<div
  aria-busy="true"
  class="feather-progress-container indeterminate"
>
  <span
    class="reduced-motion-caption"
  >
    Loading...
  </span>
  <div
    class="indeterminate-container lg"
  >
    <svg
      viewBox="0 0 100 100"
    >
      <clippath
        id="clip"
      >
        <path
          d="M 50 0 a 50 50 0 0 1 0 100 50 50 0 0 1 1 -100 v 10 a 22 22 0 0 0 0 80 22 22 0 0 0 1 -80"
        />
      </clippath>
      <foreignobject
        clip-path="url(#clip)"
        height="100"
        width="100"
        x="0"
        y="0"
      >
        <div
          class="track"
        />
      </foreignobject>
      <g
        class="indicator"
      >
        <foreignobject
          clip-path="url(#clip)"
          height="100"
          width="100"
          x="0"
          y="0"
        >
          <div
            class="gradient"
          />
        </foreignobject>
        <!-- &lt;circle class=&quot;tip&quot; r=&quot;5.15&quot; cx=&quot;51&quot; cy=&quot;5&quot;&gt;&lt;/circle&gt; -->
        <rect
          class="tip"
          height="10"
          width="3"
          x="50"
        />
      </g>
    </svg>
  </div>
  <!--v-if-->
  <progress
    max="100"
  />
  <span
    aria-live="assertive"
    class="announce"
  />
  
  <!--v-if-->
  
</div>
`;
